pkg/controller/elasticsearch/reconcile/status.go

```diff
@@ -105,7 +105,7 @@ func (u *UpscaleReporter) HasPendingNewNodes() bool {
 // Merge creates a new upscale status using the reported upscale status and an existing upscale status.
 func (u *UpscaleReporter) Merge(other esv1.UpscaleOperation) esv1.UpscaleOperation {
 	upscaleOperation := other.DeepCopy()
-	if u == nil {
+	if u == nil || (len(u.nodes)+len(other.Nodes) == 0) {
```

Comment:
I think one of the original idea was to always update `LastUpdatedTime`, which is no more the case here. Also having an empty slice of nodes instead of `nil` was a clear message to the user that the status has been effectively processed, making it explicit that there is no nodes involved in an operation. For example after an upgrade:

* Original status:
```json
  "inProgressOperations": {
    "downscale": {
      "lastUpdatedTime": "2022-03-30T05:33:33Z",
      "nodes": []
    },
    "upgrade": {
      "lastUpdatedTime": "2022-03-30T05:37:51Z",
      "nodes": []
    },
    "upscale": {
      "lastUpdatedTime": "2022-03-30T05:33:33Z",
      "nodes": []
    }
  }
```

* With this PR:
```json
  "inProgressOperations": {
    "downscale": {},
    "upgrade": {
      "lastUpdatedTime": "2022-03-30T05:27:48Z"
    },
    "upscale": {}
  }
```

I think I'm fine with the new approach though (but I still prefer the original one tbh). In that case I think we can simplify the following code since the lengths of of the slices have already been tested:

```go
	if (d.nodes != nil && !reflect.DeepEqual(nodes, other.Nodes)) || downscaleOperation.Nodes == nil {
		downscaleOperation.Nodes = nodes
		downscaleOperation.LastUpdatedTime = metav1.Now()
	}
```

Reply:
I too prefer the original approach. Here is another version to avoid converting the nil slices into empty slices and keeping the logic behind `LastUpdatedTime`: b99f0f9e5.